### DISCUSSION TITLE
SSO with Azure AD

### DISCUSSION THREAD

**Discussion by arul24ar** [provectus/kafka-ui#3333] [Q&A] [answered]
Enabled AUTH_TYPE: OAUTH2 and configured all the registration and provider details below. But still I get the error as "java.lang.IllegalArgumentException: registrations cannot be null or empty"

Below are my configs:
  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_CLIENTNAME: "azure"
  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_CLIENTID: xxxxxxxxxxx
  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_CLIENTSECRET:  xxxxxxxxxxxxx
  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_SCOPE: "https://graph.microsoft.com/User:Read"
  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_PROVIDER: "AZURE"
  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_REDIRECT_URI: "xxxxxxxxxxxx"
  SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_AUTHORIZATION_GRANT_TYPE: authorization_code

**Comment by Haarolean** [comment, marked as answer]
See [breaking changes](https://github.com/provectus/kafka-ui/releases/tag/v0.5.0) for 0.5.0

**Reply by arul24ar** [reply]
Thank you for the response, We are able integrate with Azure AD. Couple of questions I have on the below topic. 

1. User Profile after login is not showing up the email or ID in the profile, It is showing random values on the profile. Refer the attachment:
![Screen Shot 2023-02-14 at 12 39 25 PM](https://user-images.githubusercontent.com/36252442/218842528-082a7d23-8e2d-4b82-89ce-a859c6030416.png)

2. RBAC - I don't not find the support for OAUTH_AUTH0, I do find the support for google, cognito and github Do we expect anything in near future to support OAUTH_AUTH0 ?

**Reply by Haarolean** [reply]
Glad it works.
1. you have to adjust `user-name-attribute` property' value.
2. Yes, it's planned -- #2844
